sdk/cosmos/azure_data_cosmos/src/models/container_definition.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use serde::Serialize;
+
+use crate::models::{IndexingPolicy, PartitionKeyDefinition};
+
+/// Describes the properties used to create a new container.
+#[derive(Clone, Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ContainerDefinition {
```

Comment:
Aren't there more things here - throughput options, VectorEmbeddingPolicy, etc?

Ok to start with this and add more if we can do it non-breaking though.

Reply:
I've added the remaining things. Basically, I removed `ContainerDefinition` and am using `ContainerProperties` for both the Create request body and the Read response body. This mirrors what we do in Go.

The `SystemProperties` struct we use to contain things like `_etag` is still marked `#[non_exhaustive]` though, which means it's not constructable using struct syntax (`SystemProperties {...}` will produce an error outside the defining crate). However, the `SystemProperties` crate implements `Default`, so `SystemProperties::default()` can be used to construct an "empty" value. 

I think this benefits us, since we don't want users to try to _set_ those values anyway (even if they're ignored by the server). So, to construct a `ContainerProperties`, you have to use this syntax:

```rust
db_client.create_container(
    ContainerProperties {
        id: "MyContainer".to_string(),
        partition_key: "/partitionKey".into(),
        ..Default::default() // Set all remaining fields to their default
    }, None);
```

Using `..Default::default()` is allowed, because it uses `SystemProperties::default()` (provided by it's own implementation of `Default`) to construct that object, which results in all the properties being `None`. We **already** want users to use this syntax to create _all_ of our request models, so that we can safely add new properties later. So, this should work out fine.

I added a test which should serve to make sure we don't add non-optional properties to `ContainerProperties` in the future, and retain the same JSON serialization whenever a user uses the basic syntax above.

Reply:
> throughput options

These are actually specified as headers and appear on the `options` type in other SDKs like Go, so they'll come later.

Reply:
This looks great.  One thing that occurred to me is that we might need to do something extra work with deserialization of `ContainerProperties` and its nested types.  To support older SDKs talking to newer versions of the service, we might need to be able to roundtrip `ContainerProperties` for things we don't know about in the SDK. The scenario is that I want to modify some property, so I query the current properties, make some change, and then set the properties.  If we don't searialize everything we get back, we might lose options.

AFAIK, .NET handles this with [`[JsonExtensionData]`](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/f88605c207801bf305e9dc890c026e208cb4d041/Microsoft.Azure.Cosmos/src/Resource/Settings/ContainerProperties.cs#L87C1-L92C88)

Reply:
I believe serde does support a similar mechanism, I'll look in to it (but might merge this first, as you suggest in your review comment)

Reply:
Well, unfortunately I'm wrong. Serde doesn't provide a built-in way to do this. We do have some options to implement it (dropping to manually implementing `Serialize` and `Deserialize` if necessary) but it doesn't look like there's a built-in feature for this.

I don't believe all the other SDKs do this, I know .NET has `[JsonExtendedData]` but I don't see similar things for Go, so I think it's reasonable for us to consider it later on.

Reply:
Ok, actually it looks like `#[serde(flatten)]` can be used for this purpose: https://serde.rs/attr-flatten.html#capture-additional-fields

Still, going to hold off for now, but at least I have an idea of how we could do it.